terraform/dns.tf

```diff
@@ -0,0 +1,26 @@
+data "azurerm_dns_zone" "cluster_dns_zone" {
+  name                = var.cluster_dns_zone
+  resource_group_name = var.cluster_dns_resource_group_name
+}
+
+data "kubernetes_service" "default" {
+  metadata {
+    name = "ingress-nginx-controller"
+  }
+
+  depends_on = [
+    helm_release.ingress-nginx
+    ]
+
+}
+
+resource "azurerm_dns_a_record" "cluster_a_record" {
+
+  count = var.add_dns
```

Comment:
Since this depends on cluster_dns_zone existence, could we remove add_dns variable and replace the condition with something like:
```
count = var.cluster_dns_zone != null ? 1 : 0
```

Reply:
updated condition